tpm2/tpm2.go

```diff
@@ -124,11 +126,13 @@ func decodeReadPCRs(in []byte) (map[int][]byte, error) {
 		return nil, err
 	}
 
-	sel, err := decodeTPMLPCRSelection(buf)
+	sels, err := decodeTPMLPCRSelection(buf)
 	if err != nil {
 		return nil, fmt.Errorf("decoding TPML_PCR_SELECTION: %v", err)
 	}
 
+	sel := sels[0]
```

Comment:
Check len(sels) here as before.

Since this is needed multiple times, maybe add a helper func:
```
func decodeOneTPMLPCRSelection(buf *bytes.Buffer) (PCRSelection, error) {
  sels, err := decodeTPMLPCRSelection(buf)
  if err != nil {
    return PCRSelction{}, err
  }
  if len(sel) != 1 {
    return nil, fmt.Errorf("got %d TPMS_PCR_SELECTION items in TPML_PCR_SELECTION, expected 1", len(sel))
  }
  return sel[0], nil
}
```